Azure/communication-ui-library-android
# [Features] Hold Support and AudioFocus

## Purpose

* Hold Feature (Invoked by loss of AudioFocus, i.e. PTSN call or other audio producer app that uses the API, e.g. games, music players, other calling apps).

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Start a meeting, test following scenarios
* Incoming PTSN call should put call on hold
* Ongoing other call should be handled (either fail to start, or other call should go on hold or end)
* Test other calling app interfering (E.g. Duo, Skype, Facebook)
* Test Spotify mutes on call started
* GPS Announcements, etc

Track any behaviours so we can document. How this behaves depends on how other apps fulfill the contract.

Intention is to hook this up to AudioFocus to auto-hold in situations where we lose focus.

## What to Check
Verify that the following are valid
* ...

## Other Information
